### ISSUE TITLE
Support priority, tolerations and affinity of infrastructure

### ISSUE THREAD

**Issue by lulf** [EnMasseProject/enmasse#2474]
**Overview**

At present, all components have a default affinity setting. Operator components have affinity to nodes with operator-infra, and messaging components have affinity towards nodes with messaging-infra.

When clusters are shared with other applications, it is desirable to control more settings of the deployed pods through the infra configs. 

* Taints and tolerations are a way to ensure that nodes are reserved for some components only: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
* Affinity settings are a way to prefer and enforce where a component is scheduled and used for supporting multiple availability zones: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
* Pod priority can be used to assign pods to a priority class that dictates eviction policy of pods of that class: https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/
* Additional labels for pods to be used by above settings

**Design**

The InfraConfigs should expose these settings for each component (showing only 'router' for brevity):

```
kind: StandardInfraConfig
spec:
  router:
    affinity: // Follows schema of https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#affinity-v1-core
    tolerations: // Follows schema of https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#toleration-v1-core
    priorityClassName:  // Just a string value
    labels: // Map of labels that are applied to components. Note that any internal labels will override values specified here.
```

**Tasklist**

* [ ] Add fields to BrokeredInfraConfig and StandardInfraConfig schema
* [ ] Copy values from the fields in to statefulset/deployment pod template when creating address space
* [ ] Copy values from the fields in to statefulset pod template when creating broker

**Comment by lulf** [comment]
@rgodfrey @robshelly @k-wall  One thing I'm thinking of is to make the format follow the kubernetes structure more directly. I.e.:

```
spec:
  router:
    pod:
      metadata:
        labels:
           mylabel: mykey
      spec:
        affinity:
        tolerations:
        priorityClassName:
        resources:
```

Although the schema follows the Kubernetes podtemplate specification, we can choose which settings we support to be applied.

Another advantage would be that settings that are not directly related to the pod, I.e. router settings  would be left directly under 'router', which gives a nice separation of Kubernetes and router concepts.

I'm a bit divided on the approach though, because it does make the whole structure a bit more verbose than strictly necessary.

**Comment by rgodfrey** [comment]
@lulf I think I prefer the second suggestion (adding under a pod: entry), we'd have this under router and broker I assume?  For the brokered config are we going to also allow these?

**Comment by lulf** [comment]
Under router broker and admin. They also make sense for brokered config I think

**Comment by rgodfrey** [comment]
@lulf ok - as we've discussed elsewhere, at some point we're going to have to think about how queue scheduling for sharded queues will ensure that the intent of (anti-) affinity is preserved when we move to a model where we lose the distinction between pooled and sharded and number of shards is settable independently (at that point do we just create distinct stateful sets for each shard size I wonder), but for now I think this is good

**Comment by lulf** [comment]
@rgodfrey In #2482 i've changed it to use podTemplate which is matches better with what it is i think

**Comment by lulf** [comment]
Implemented in #2482
